src/App.tsx

```diff
@@ -1,24 +1,217 @@
 import { createSignal } from "solid-js";
 
+const WORDS = ["SOLID", "JAVASCRIPT", "TAILWIND", "VITE", "SEARCH"];
+const GRID_SIZE = 10;
+
+function getRandomInt(max: number) {
+	return Math.floor(Math.random() * max);
+}
+
+function createEmptyGrid(size: number) {
+	return Array.from({ length: size }, () =>
+		Array.from({ length: size }, () => ""),
+	);
+}
+
+function placeWords(grid: string[][], words: string[]) {
+	const directions = [
+		[0, 1], // right
+		[1, 0], // down
+		[1, 1], // diagonal down-right
+		[-1, 1], // diagonal up-right
+	];
+	for (const word of words) {
+		let placed = false;
+		for (let attempts = 0; attempts < 100 && !placed; attempts++) {
+			const dir = directions[getRandomInt(directions.length)];
+			const row = getRandomInt(GRID_SIZE);
+			const col = getRandomInt(GRID_SIZE);
+			let fits = true;
+			for (let i = 0; i < word.length; i++) {
+				const r = row + dir[0] * i;
+				const c = col + dir[1] * i;
+				if (
+					r < 0 ||
+					r >= GRID_SIZE ||
+					c < 0 ||
+					c >= GRID_SIZE ||
+					(grid[r][c] && grid[r][c] !== word[i])
+				) {
+					fits = false;
+					break;
+				}
+			}
+			if (fits) {
+				for (let i = 0; i < word.length; i++) {
+					const r = row + dir[0] * i;
+					const c = col + dir[1] * i;
+					grid[r][c] = word[i];
+				}
+				placed = true;
+			}
+		}
+	}
+}
+
+function fillGridRandom(grid: string[][]) {
+	const letters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+	for (let r = 0; r < GRID_SIZE; r++) {
+		for (let c = 0; c < GRID_SIZE; c++) {
+			if (!grid[r][c]) {
+				grid[r][c] = letters[getRandomInt(letters.length)];
+			}
+		}
+	}
+}
+
+function getDefaultGrid() {
+	const g = createEmptyGrid(GRID_SIZE);
+	placeWords(g, WORDS);
+	fillGridRandom(g);
+	return g;
+}
+
 function App() {
-	const [count, setCount] = createSignal(0);
+	const [selected, setSelected] = createSignal<[number, number][]>([]);
+	const [found, setFound] = createSignal<string[]>([]);
+	const [grid, setGrid] = createSignal<string[][]>(getDefaultGrid());
+
+	function handleCellClick(row: number, col: number) {
+		const sel: [number, number][] = [...selected(), [row, col]];
+		setSelected(sel);
+		const word = sel.map(([r, c]) => grid()[r][c]).join("");
+		const reversed = word.split("").reverse().join("");
+		const matched = WORDS.find((w) => w === word || w === reversed);
+		if (matched && !found().includes(matched)) {
+			setFound([...found(), matched]);
+			setSelected([]);
+		} else if (word.length > Math.max(...WORDS.map((w) => w.length))) {
```

Comment:
having to click letters until letter count is greater than any word to find, in order to clear the selection, is not very intuitive